### PR TITLE
statistics: add more comments for the histogram bucket

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: None

Problem Summary:

We need to add more comments to the histogram bucket structure to make it easier to understand.

### What changed and how does it work?

- Added some comments for the fields from the bucket struct.
- Added some comments for building histogram procedure.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [x] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

File: pkg/statistics/histogram.go [hunk 1]
```diff
@@ -87,9 +87,13 @@ const EmptyHistogramSize = int64(unsafe.Sizeof(Histogram{}))
 
 // Bucket store the bucket count and repeat.
 type Bucket struct {
-	Count  int64
+	// Count is the number of items in the bucket.
+	Count int64
+	// Repeat is the number of repeats of the upper bound of the bucket.
```
Comment (AilinKid) [inline pkg/statistics/histogram.go:92]: how about using **_right_** bound of this bucket range [x,**_y_**], more visually clear

File: pkg/statistics/histogram.go [hunk 2]
```diff
@@ -87,9 +87,13 @@ const EmptyHistogramSize = int64(unsafe.Sizeof(Histogram{}))
 
 // Bucket store the bucket count and repeat.
 type Bucket struct {
-	Count  int64
+	// Count is the number of items in the bucket.
```
Comment (AilinKid) [inline pkg/statistics/histogram.go:90]:
```suggestion
	// Count is the number of items till this bucket.
```

**Review by hawkingrei — APPROVED** [review]

**Review by AilinKid — APPROVED** [review]
LGTM

**Pull request merged**